src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/memory/FileMapInfo.java

```diff
@@ -125,8 +126,10 @@ private static void populateMetadataTypeArray(TypeDataBase db) {
     metadataTypeArray[4] = db.lookupType("InstanceRefKlass");
     metadataTypeArray[5] = db.lookupType("InstanceStackChunkKlass");
     metadataTypeArray[6] = db.lookupType("Method");
-    metadataTypeArray[7] = db.lookupType("ObjArrayKlass");
-    metadataTypeArray[8] = db.lookupType("TypeArrayKlass");
+    metadataTypeArray[9] = db.lookupType("MethodData");
```

Comment:
It looks like MethodData inheriting from Metadata is not a new change, but has always been the case. I'm surprised this didn't cause any test failures before your changes. Did you end up with test failures after your changes?

Reply:
Honestly I don't remember, I think @iklam did these changes.